### ISSUE TITLE
Is there visual flash effect for vim editing

### ISSUE THREAD

**Issue by deleisha** [neovim/neovim#4182]
Replicated from [StackOverflow](http://stackoverflow.com/questions/35171726/is-there-visual-flash-effect-for-editing)

What we are looking is to visually flash briefly the affected areas of vim editing in normal mode. For example, when editing

```C
if (true) {
    //line to be deleted
}
```

if we do dd on //line to be deleted, this affected area should be flashed before deleting, The same we can do using Vd. What we are looking is same effect as that of Vd using dd. This should work for all editing operations like c, y etc

May be the effect can be controlled using a configuration option

**Comment by teto** [comment]
You can use vim-operator-flashy for that:
Plug 'haya14busa/vim-operator-flashy' " Flash selection on copy

**Comment by justinmk** [comment]
https://github.com/inside/vim-search-pulse works pretty well, it could be adapted for this purpose.

**Comment by deleisha** [comment]
Flashy works only for copy.  Will try search pulse

**Comment by deleisha** [comment]
Well none of them actually do what we are looking for. They work for some particular cases like `searching` or `yanking`. 

Going by their approaches, It will require to reimplement for all `operator`. 

What we are looking is a way that can be `plug in the effect` for every editing command.

**Comment by deleisha** [comment]
@justinmk Would you reopen this. 
This request will be a good value add to neovim.

**Comment by wsdjeg** [comment]
@deleisha I do not think it is a good thing,if you type `dd`,why need we go to visual mode?it is just delete current line.you can use `Vd`,I think it is same thing when you first time to use `j` or `k` to move cursor when you used use `up` and `down`.

**Comment by justinmk** [comment]
https://github.com/neovim/neovim/issues/16296 is related. But for this issue, we can do something pretty inexpensive:

# Proposal

- Ensure that the `TextChanged` event indicates which region was changed, similar to `TextYankPost`, `TextPutPost` ([ref](https://github.com/neovim/neovim/pull/39741))
- Then `vim.hl.hl_op()` can highlight it. https://github.com/neovim/neovim/pull/39777

**Comment by mCaballero1224** [comment]
Hello, I would like to be assigned to this if possible. I'm looking to start contributing to Neovim and since this is listed as low complexity I figured it'd be a way for me to ease into that.

**Comment by justinmk** [comment]
You could continue https://github.com/neovim/neovim/pull/34542 which was closed.

**Comment by daigo-suhara** [comment]
I'm a student interested in participating in GSoC 2026.
I’d like to work on highlighting the parts to be deleted as my first contribution to neovim.

**Comment by jakobferguson** [comment]
From my understanding the TextChanged event occurs after the text is removed, which would prevent us from having this feature active for deletes and cuts. Does that sound accurate? Would an acceptable implementation be creating an event for after a command has been issued but prior to operating on text?

**Comment by justinmk** [comment]
Not accepting implementations for this. To achieve visual-first editing https://github.com/neovim/neovim/issues/16296 , that will be enabled by the upcoming multicursor engine.

Otherwise, it's either TextChanged or "wontfix".
